### PR TITLE
US75544 - Expired password Readme fix

### PR DESCRIPTION
**Pull request by ikemsley**
**Problem**

If a user's password is expired the "Key creation" command described in the GD README says "authentication failure" which is misleading.

**Solution**

If AIMs returns an error message in the body then report the error to the user.

### CODE REVIEW

**Comment by ikemsley** [comment]
@kkuzmin @tomdos @alexturkin Please CR.  Thanks.

**Comment by kkuzmin** [comment]
@msayler @rhosmo @Shaddakim please take a look

File: cfn/README-GD.md [hunk 1]
```diff
@@ -56,8 +56,9 @@ From the bash command line, type the following commands, where `<email address>`
...
+auth=$(curl -X POST -s -u $AL_USERNAME https://api.global-integration.product.dev.alertlogic.com/aims/v1/authenticate); export AL_ERROR=$(echo $auth | jq -r '.error // ""'); export AL_ACCOUNT_ID=$(echo $auth | jq -r '.authentication.account.id'); export AL_USER_ID=$(echo $auth | jq -r '.authentication.user.id'); export AL_TOKEN=$(echo $auth | jq -r '.authentication.token'); if [ -n "$AL_ERROR" -o -z "$AL_TOKEN" ]; then echo "Authentication failure - $AL_ERROR "; else roles=$(curl -s -X GET -H "x-aims-auth-token: $AL_TOKEN" https://api.global-integration.product.dev.alertlogic.com/aims/v1/$AL_ACCOUNT_ID/users/$AL_USER_ID/roles | jq -r '.roles[].name'); if [ "$roles" != "Administrator" ]; then echo "The $AL_USERNAME doesn’t have Administrator role. Assigned role is '$roles'"; else curl -s -X POST -H "x-aims-auth-token: $AL_TOKEN" https://api.global-integration.product.dev.alertlogic.com/aims/v1/$AL_ACCOUNT_ID/users/$AL_USER_ID/access_keys | jq .; fi; fi; unset AL_USERNAME;
```
Comment (kkuzmin) [inline cfn/README-GD.md]: @ikemsley please fix the URLs `api.global-integration.product.dev.alertlogic.com`

Comment (ikemsley) [reply cfn/README-GD.md]: @kkuzmin Whoops!  Good catch!

**Comment by msayler** [comment]
verified. What about Windows?

**Comment by rhosmo** [comment]
New text incorporated into Webhelp.

**Comment by ikemsley** [comment]
@msayler @kkuzmin 

Turns out with windows the call fails and displays the AIMs response anyway.  Bit messy, but it looks like this:

```
Please enter your Alert Logic Cloud Insight email address and password
User: ikemsley@hotmail.com
Password for user ikemsley@hotmail.com: *************

Invoke-RestMethod : {"error":"password_expired"}
At line:1 char:418
+ ... ds; $AUTH = Invoke-RestMethod -Method Post -Headers @{"Authorization" ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : InvalidOperation: (Method: POST, R...rShell/6.0.0
}:HttpRequestMessage) [Invoke-RestMethod], HttpResponseException
+ FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeRestMethodCommand
```

So I don't think this particular issue applies to windows.

**Comment by msayler** [comment]
Makes sense, thanks.

**Pull request merged**